src/verify/starknet.ts

```diff
@@ -67,16 +67,24 @@ export default async function verify(
   network: NetworkType = 'SN_MAIN',
   options: ProviderOptions = {}
 ): Promise<boolean> {
-  const contractAccount = new Contract(
-    ABI,
-    address,
-    getProvider(network, options)
-  );
+  try {
+    const contractAccount = new Contract(
+      ABI,
+      address,
+      getProvider(network, options)
+    );
+
+    await contractAccount.is_valid_signature(
+      getHash(data, address),
+      sig.slice(-2)
+    );
 
-  await contractAccount.isValidSignature(getHash(data, address), [
-    sig[0],
-    sig[1]
-  ]);
+    return true;
```

Comment:
Looks like we are checking `is_valid_signature` and irrespective of the response, we return true, maybe that's the reason it accepts testnet signatures also?

Reply:
The function is always throwing and error when not `true`

Reply:
Added a check to compare returned result before returning `true`